### ISSUE TITLE
write_binary always uses layers :raw:encoding('latin-1'), should be just :raw.

### ISSUE THREAD

**Issue by dtko** [Leont/file-slurp-sane#4]
The write_binary code looks like:
`sub write_binary {
	return write_text(@_[0,1], 'latin-1');
}`
so passes an encoding of 'latin-1', with a hyphen.

write_text calls _text_layers which uses encoding and crlf to determine the encoding layer.

The relevant part of _text layers is:
`	if ($encoding =~ /^(latin|iso-8859-)1$/i) {
		return $crlf ? ':unix:crlf' : ':raw';
	}`

That matches "latin1" but not "latin-1".  write_binary does not write using ':raw', but instead ":raw:encoding('latin-1')".

This produces a "wide character" warning for each non-ascii line written.

To fix, either change your regex, maybe `/^(latin-?|iso-8859-)1$/i`, or change write_binary to: `return write_text(@_[0,1], 'latin1');` , no hyphen.

I could branch and create a pull request but that seems silly for a 1-character fix.

**Comment by Leont** [comment]
> The write_binary code looks like:
> `sub write_binary { return write_text(@_[0,1], 'latin-1'); }`
> so passes an encoding of 'latin-1', with a hyphen.
> 
> `write_text calls _text_layers which uses encoding and crlf to determine the encoding layer.`
> 
> The relevant part of _text layers is:
> `if ($encoding =~ /^(latin|iso-8859-)1$/i) { return $crlf ? ':unix:crlf' : ':raw'; }`
>
> That matches "latin1" but not "latin-1". write_binary does not write using ':raw', but instead ":raw:encoding('latin-1')".

Yes, that is a bug.

> This produces a "wide character" warning for each non-ascii line written.
>
> To fix, either change your regex, maybe /^(latin-?|iso-8859-)1$/i, or change write_binary to: return write_text(@_[0,1], 'latin1'); , no hyphen.
>
> I could branch and create a pull request but that seems silly for a 1-character fix.

Yeah that first solution is clearly what I'd want there, though this is making me reconsider some implementation details of the module (should I make the binary/latin-1 distinction sharper?)

**Comment by dtko** [comment]
I would make that distinction clear.  If I understand correctly, you never want to set any encoding or any crlf character with write_binary, so that method could be a simple and direct:
```
sub write_binary {
	my $filename = $_[0];
	open my $fh, ">:raw", $filename or croak "Couldn't open $filename: $!";
	print $fh $_[1] or croak "Couldn't write to $filename: $!";
	close $fh or croak "Couldn't write to $filename: $!";
	return;
}
```
It duplicates a few lines of code but you avoid going through _text_layers entirely.

**Comment by Leont** [comment]
Fixed in 0.011
